### PR TITLE
Update cargo install instructions to use dev.net branch

### PR DESCRIPTION
**Pull request by velvia**

### CODE REVIEW

File: doc/src/build/install.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ version 1.60.0 or higher in order to build and install Sui on your machine.
 To develop in Sui, you will need the Sui binaries. After installing `cargo`, run:
 
 ```shell
-$ cargo install --git https://github.com/MystenLabs/sui.git sui
+$ cargo install --locked --git https://github.com/MystenLabs/sui.git --branch "devnet"
```
Comment (666lcz) [inline doc/src/build/install.md:15]:
```suggestion
$ cargo install --locked --git https://github.com/MystenLabs/sui.git --branch "devnet" sui
```

**Review by longbowlu — APPROVED** [review]

**Comment by 666lcz** [comment]
Looks good to me. Please let @Clay-Mysten take a look before merging

**Comment by Clay-Mysten** [comment]
Is this going to work for non-DevNet users?

**Comment by velvia** [comment]
@Clay-Mysten yeah it will absolutely work, though they may not have the latest and greatest changes.

**Review by Clay-Mysten — APPROVED** [review]

**Review by bmwill — APPROVED** [review]

**Pull request merged**